cli/config.go

```diff
@@ -4,65 +4,209 @@
 package cli
 
 import (
-	"fmt"
+	"io"
 	"os"
+	"reflect"
+	"strconv"
 
 	"github.com/mainflux/mainflux/pkg/errors"
+	mfxsdk "github.com/mainflux/mainflux/pkg/sdk/go"
 	"github.com/pelletier/go-toml"
+	"github.com/spf13/cobra"
 )
 
-type Config struct {
-	Offset    uint   `toml:"offset"`
-	Limit     uint   `toml:"limit"`
+type remotes struct {
+	ThingsURL       string `toml:"things_url"`
+	UsersURL        string `toml:"users_url"`
+	ReaderURL       string `toml:"reader_url"`
+	HTTPAdapterURL  string `toml:"http_adapter_url"`
+	BootstrapURL    string `toml:"bootstrap_url"`
+	CertsURL        string `toml:"certs_url"`
+	MsgContentType  string `toml:"msg_content_type"`
+	TLSVerification bool   `toml:"tls_verification"`
+}
+
+type filter struct {
+	Offset    string `toml:"offset"`
+	Limit     string `toml:"limit"`
+	RawOutput string `toml:"raw_output"`
 	Name      string `toml:"name"`
-	RawOutput bool   `toml:"raw_output"`
+	Contact   string `toml:"contact"`
+	Email     string `toml:"email"`
+	Metadata  string `toml:"metadata"`
 }
 
-// read - retrieve config from a file.
-func read(file string) (Config, error) {
-	data, err := os.ReadFile(file)
-	c := Config{}
+type channel struct {
+	Status string `toml:"status"`
+	State  string `toml:"state"`
+	Topic  string `toml:"topic"`
+}
+
+type config struct {
+	Remotes remotes `toml:"remotes"`
+	Filter  filter  `toml:"filter"`
+	Channel channel `toml:"channel"`
+}
+
+func read(file string) (config, error) {
+	c := config{}
+	data, err := os.Open(file)
+	if err != nil {
+		return c, errors.Wrap(errors.New("Failed to read config file."), err)
+	}
+	defer data.Close()
+
+	buf, err := io.ReadAll(data)
 	if err != nil {
-		return c, errors.New(fmt.Sprintf("failed to read config file: %s", err))
+		return c, errors.Wrap(errors.New("Failed to read config file."), err)
 	}
 
-	if err := toml.Unmarshal(data, &c); err != nil {
-		return Config{}, errors.New(fmt.Sprintf("failed to unmarshal config TOML: %s", err))
+	if err := toml.Unmarshal(buf, &c); err != nil {
+		return config{}, errors.Wrap(errors.New("failed to Unmarshall config TOML."), err)
 	}
+
 	return c, nil
 }
 
-func ParseConfig() error {
+func ParseConfig(sdkConf mfxsdk.Config) mfxsdk.Config {
```

Comment:
Wrong error is being outputed 
![image](https://github.com/mainflux/mainflux/assets/28790446/5a22369c-6126-43b2-9b57-2be3112fac93)

Reply:
Solved